### DISCUSSION TITLE
How to get user game status?

### DISCUSSION THREAD

**Discussion by Tapicz** [Rapptz/discord.py#6380] [Q&A]
Hey, i want to make bot check if any user on server is playing game, and if yes, then what game. I don't really have clue how to get this working. Please help

**Comment by bandwiches** [comment]
This is found via Member attributes `activity` or `activities`

[Member Object](https://discordpy.readthedocs.io/en/latest/api.html?highlight=game#member)
[Activity Data Class](https://discordpy.readthedocs.io/en/latest/api.html?highlight=game#activity)
[Game Data Class](https://discordpy.readthedocs.io/en/latest/api.html?highlight=game#game)

**Reply by Tapicz** [reply]
Is there any chance, that you have reference code?

**Reply by bandwiches** [reply]
Hopefully this can get you started

```
@bot.listener
async def on_member_event(self, before: discord.Member, after: discord.Member):
  print(before.activites)
  print(after.activities)
```

**Reply by tk421bsod** [reply]
Adding on to this, you need to enable the presences intent before you can access activities. Read [the docs](https://discordpy.readthedocs.io/en/latest/intents.html) for intents for information on how to do this.

**Reply by Tapicz** [reply]
> 
> 
> Hopefully this can get you started
> 
> ```
> @bot.listener
> async def on_member_event(self, before: discord.Member, after: discord.Member):
>   print(before.activites)
>   print(after.activities)
> ```

Traceback (most recent call last):
  File "D:\Programming\Python\Bot\main.py", line 8, in <module>
    @bot.listener
AttributeError: module 'discord.ext.commands.bot' has no attribute 'listener'

what i must import, to get this work?

**Reply by bandwiches** [reply]
My syntax was wrong, it should be `@bot.listen()`.  Depending on how your bot is built, her are the resources: [Bot.listen()](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.listen) [Cog.listener()](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Cog.listener)

**Reply by Tapicz** [reply]
> 
> 
> My syntax was wrong, it should be `@bot.listen()`. Depending on how your bot is built, her are the resources: [Bot.listen()](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.listen) [Cog.listener()](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Cog.listener)

```
import discord
from discord.ext import commands
from discord.ext.commands import bot

client = commands.Bot(command_prefix=".")


@bot.listen()
async def on_member_event(self, before: discord.Member, after: discord.Member):
    print(before.activites)
    print(after.activities)


@client.event
async def on_ready():
    print("Bot is ready")


client.run(" * Token * ")

```

Error:

```
Traceback (most recent call last):
  File "D:\Programming\Python\Bot\main.py", line 8, in <module>
    @bot.listen()
AttributeError: module 'discord.ext.commands.bot' has no attribute 'listen'
```

Still not working :(

**Reply by bandwiches** [reply]
Your `bot` is actually `client` based on your code.  See how you have `@client.event` setup?  It's just like that except in your case it should be: `@client.lisen()`.  Don't forget the `()`.  For reference - [Bot.listen](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.listen)
